### DISCUSSION TITLE
Variable results based on roll

### DISCUSSION THREAD

**Discussion by FearEMTKelly** [AngryBeaver/beavers-crafting#133] [Ideas]
I'm not sure if this already exists. I am trying to set up a cooking mechanic for my party where the results of their cooking check dictates how good their food is. I would like to see if it's possible as is or maybe possible to add results that differ based on the role made by the player?

**Comment by AngryBeaver** [comment]
It is no feature that will come with some fancy click ui that would be a hell to implement. 
As usually when something is too complicated for me to implement it generic with fancy ui you still can do it yourself by adding a macro to the recipe.

But this is advanced stuff for people with some development experience, i won't help each and every one to code his perfect macro in there :-). 
As startup you can look up here but i think it is a bit outdated the (interfaces had changed a bit) use the console.log syntax to get the current interfaces.
https://github.com/AngryBeaver/beavers-crafting/blob/main/macros.md


As to why is this complicated to code in ui. 
1.) you can have multiple rolls not only one
1.a) i would need to setup that each roll would change the outcome.
1.b) what if you want your result to changed based on combined rolls e.g. first roll less then second roll more then ??-> matrix of possibilities !!!
2.) you might have recipes with no rolls at all.
3.) you might want this feature for the overall win or fails not for a specific roll dc.
4.) some systems do not return rolled dc (<- this is typically dnd variant like) (some just return success or failure)

so makeing a macro for just your recipe usecase is much simpler for you then for me coding a feature for all possible usecases.

## tldr:
you can do so with macros:
https://github.com/AngryBeaver/beavers-crafting/blob/main/macros.md

**Comment by FearEMTKelly** [comment]
Thank you for replying.  I had no idea it would be that crazy.  Thank you
for what you do and all the work you put into it!
